modules/carousel/jetpack-carousel.php

```diff
@@ -543,6 +543,7 @@ class_exists( 'Jetpack_AMP_Support' )
 			foreach ( (array) $extra_data as $data_key => $data_values ) {
 				$html = str_replace( '<div ', '<div ' . esc_attr( $data_key ) . "='" . json_encode( $data_values ) . "' ", $html );
 				$html = str_replace( '<ul class="wp-block-gallery', '<ul ' . esc_attr( $data_key ) . "='" . json_encode( $data_values ) . "' class=\"wp-block-gallery", $html );
+				$html = str_replace( '<ul class="blocks-gallery-grid', '<ul ' . esc_attr( $data_key ) . "='" . json_encode( $data_values ) . "' class=\"blocks-gallery-grid", $html );
```

Comment:
I noticed a small problem with the str_replace() call. Your change didn't introduce the problem, but maybe it could be fixed in this PR. This str_replace()will add only the first $data_key/$data_value pair in $extra_data to the ul element. The first loop changes the element to <ul key=‘value’ class=“blocks-gallery-grid. So, the string <ul class="blocks-gallery-grid no longer exists.

Reply:
Oh wow, great catch! It's unrelated to this fix and will need appropriate testing and time, so I think it's best to leave it separate; filed here: https://github.com/Automattic/jetpack/issues/13604